listenbrainz/webserver/views/playlist_api.py

```diff
@@ -0,0 +1,516 @@
+        from_value = int(data["from"])
+        to_value = int(data["to"])
+        count_value = int(data["count"])
+        if from_value < 0 or to_value < 0 or count_value < 0:
+            raise ValueError
+    except ValueError:
+        log_raise_400("move instruction values for 'from', 'to' and 'count' must all be positive integers.")
+
+
+def validate_delete_data(data):
+    """
+        Check that the passed JSON for a delete recordings endpoint call are valid. Raise 400 with
+        error message if not.
+    """
+
+    if "index" not in data or "count" not in data:
+        log_raise_400("post data for a remove instruction must include the keys 'index' and 'count'.")
+
+    try:
+        index_value = int(data["index"])
+        count_value = int(data["count"])
+        if index_value < 0 or count_value < 0:
+            raise ValueError
+    except ValueError:
+        log_raise_400("move instruction values for 'index' and 'count' must all be positive integers.")
+
+
+def fetch_playlist_recording_metadata(playlist: Playlist):
+    """
+        This interim function will soon be replaced with a more complete service layer
+    """
+
+    mbids = [{'[recording_mbid]': str(item.mbid)} for item in playlist.recordings]
+    if not mbids:
+        return
+
+    r = requests.post(RECORDING_LOOKUP_SERVER_URL, params={"count": len(mbids)}, json=mbids)
+    if r.status_code != 200:
+        current_app.logger.error("Error while fetching metadata for a playlist: %d" % r.status_code)
+        raise APIInternalServerError("Failed to fetch metadata for a playlist. Please try again.")
+
+    try:
+        rows = ujson.loads(r.text)
+    except ValueError as err:
+        current_app.logger.error("Error parse metadata for a playlist: {}".format(err))
+        raise APIInternalServerError("Failed parse fetched metadata for a playlist. Please try again.")
+
+    mbid_index = {}
+    for row in rows:
+        mbid_index[row['original_recording_mbid']] = row
+
+    for rec in playlist.recordings:
+        try:
+            row = mbid_index[str(rec.mbid)]
+        except KeyError:
+            continue
+
+        rec.artist_credit = row.get("artist_credit_name", "")
+        if "[artist_credit_mbids]" in row and not row["[artist_credit_mbids]"] is None:
+            rec.artist_mbids = [UUID(mbid) for mbid in row["[artist_credit_mbids]"]]
+        rec.title = row.get("recording_name", "")
+
+
+@playlist_api_bp.route("/create", methods=["POST", "OPTIONS"])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def create_playlist():
+    """
+    Create a playlist. The playlist must be in JSPF format with MusicBrainz extensions, which is defined
+    here: https://musicbrainz.org/doc/jspf . To create an empty playlist, you can send an empty playlist
+    with only the title field filled out. If you would like to create a playlist populated with recordings,
+    each of the track items in the playlist must have the identifier element present.
```

Comment:
should this also say that the track identifier should be a recording MBID?

Reply:
Well, a recording URI, really. I'll clarify.